### ISSUE TITLE
Security group for NodePort with AWS Load Balancer controller

### ISSUE THREAD

**Issue by nsubrahm-aws** [kubernetes-sigs/aws-load-balancer-controller#1724]
The documentation on [Application load balancing](https://docs.aws.amazon.com/eks/latest/userguide/alb-ingress.html) on Amazon EKS does not explicitly mention that the traffic type `ip` also requires the `Service` definition to use `type:NodePort`. Whereas, the [example](https://raw.githubusercontent.com/kubernetes-sigs/aws-load-balancer-controller/main/docs/examples/2048/2048_full.yaml) uses traffic type `ip` and `Service` definition explicitly uses `type:NodePort`. When I experimented with `type: ClusterIp`, and traffic type as `ip`, the Target Group reported the targets to be unhealthy.

Therefore, my first question is, does the usage of AWS Load Balancer controller mandate `Service` with `type: NodePort` _only_ ?

Given that, `NodePort` exposes a port on the underlying instance, does the controller also create a security group to limit traffic from ALB only ? The port range would have to be the entire range because, `Service` specification does not mandate `nodePort` number and relies on Kubernetes to pick one - unless specified, of course.

**Comment by kishorj** [comment]
For instance mode, Service type of `NodePort` is required. Service type `LoadBalancer` will also work since it is built on top of NodePort, but you might not need the extra LB that gets provisioned for the service.

Controller adds the security rules to the node SG to allow traffic from the ALB SG.

**Comment by nsubrahm-aws** [comment]
Thanks, @kishorj  for the response!

Can you please help compiling this type?

| Traffic Type | Service Type | Valid (Y/N) |
| ------------ | ------------- | ----------- |
| `ip` | `ClusterIp` | |
| `ip` | `LoadBalancer` | `Y` |
| `ip` | `NodePort` | `Y` |
| `instance` | `ClusterIp` | |
| `instance` | `LoadBalancer` | `Y` |
| `instance` | `NodePort` | `Y` |

I understand that, using `LoadBalancer` will spin up an instance of ALB per `Service` which might turn out to be expensive.

> Controller adds the security rules to the node SG to allow traffic from the ALB SG.

I could locate the managed security group; but, I couldn't locate the EC2 instance in the `NodeGroup` that has this security group attached. Thoughts?

**Comment by nsubrahm-aws** [comment]
#1731

**Comment by jpriebe** [comment]
> Thanks, @kishorj for the response!
> 
> Can you please help compiling this type?
> 
> Traffic Type	Service Type	Valid (Y/N)
> `ip`	`ClusterIp`	
> `ip`	`LoadBalancer`	`Y`
> `ip`	`NodePort`	`Y`
> `instance`	`ClusterIp`	
> `instance`	`LoadBalancer`	`Y`
> `instance`	`NodePort`	`Y`
> I understand that, using `LoadBalancer` will spin up an instance of ALB per `Service` which might turn out to be expensive.
> 
> > Controller adds the security rules to the node SG to allow traffic from the ALB SG.
> 
> I could locate the managed security group; but, I couldn't locate the EC2 instance in the `NodeGroup` that has this security group attached. Thoughts?

I wish the documentation had something like this table that @[nsubrahm-aws](https://github.com/nsubrahm-aws) was trying to create.  And even more, I wish it had pros/cons of each combination, and possibly a "recommended" configuration.  We found ourselves in a situation where we were using Ingress with LoadBalancer services, and it was indeed creating an NLB (clarification - not an ALB as in [nsubrahm-aws](https://github.com/nsubrahm-aws)'s post) for each service.  And with those NLBs, it was creating a bunch of security group rules against our EKS worker nodes' SG, which eventually ran up against the 60-rule limit.

So we switched those services to NodePort, which shut down the NLBs (but doesn't seem to have removed the SG rules, unfortunately).

My main point is that the docs are lacking -- the main example that gets referenced in every online article only illustrates one of the many use cases; I'd like to better understand each use case and why you would (or whether you even should) use them.
